src/main.rs

```diff
@@ -87,9 +87,30 @@ impl std::str::FromStr for Color {
     }
 }
 
+fn get_version() -> String {
+    let pkg_version = if built_info::CI_PLATFORM.is_some() && built_info::PKG_VERSION.contains('-')
+    {
+        built_info::PKG_VERSION.replace("-dev", "-nightly")
+    } else {
+        built_info::PKG_VERSION.into()
+    };
+    let commit_hash = built_info::GIT_VERSION.and_then(|v| v.split('-').last());
```

Comment:
Using `Command` + `git` would be less portable (for example, `git` command is not available most Windows machines). I agree that `libgit2` increases the build time, but it doesn't increase the final executable size because it is used only during the build.

Reply:
I think we can assume that all users compiling from source should have git and if it doesn't exist we can just skip this and place a place holder. I don't think we will ever have the case of someone building a release and not having git.

Reply:
I think we should separate the discussion about potentially moving away from `built`. Personally, I'm in favor of sticking with it (at least for the time being), but I don't think that decision needs to be made as part of this PR.

Reply:
I disagree. I do not want to assume anything, and it is possible git is not installed. _Again_, I prefer to optimize for the UX not Lucio's developer happiness.

Reply:
Not sure what UX we lose by not using built? It's completely a release thing? All we would be avoiding is having to compile libgit to fetch a commit sha. Happy to discuss this further in another issue.